storage/s3.go

```diff
@@ -374,14 +398,35 @@ func (s *S3) Put(
 		input.StorageClass = aws.String(storageClass)
 	}
 
-	_, err := s.uploader.UploadWithContext(ctx, input, func(u *s3manager.Uploader) {
+	err := setEncrytParams(&input.ServerSideEncryption, &input.SSEKMSKeyId, metadata)
+	if err != nil {
+		return err
+	}
+
+	_, err = s.uploader.UploadWithContext(ctx, input, func(u *s3manager.Uploader) {
 		u.PartSize = partSize
 		u.Concurrency = concurrency
 	})
 
 	return err
 }
 
+// validateEncryptParams validates encryption parameters and
+// returns corresponding values.
+func validateEncryptParams(metadata map[string]string) (encryptMethod, encryptKey string, err error) {
```

Comment:
Instead of doing validation against `aws:kms`, can't we just pass whatever passed as flags to SDK? `aws:kms` check is too AWS specific since we support S3 compatible services as much as we can.